### PR TITLE
feat: oauth before login

### LINKED ISSUE #231: Require confirming account ownership when starting the node or logging in
When starting the node or logging in, apart from entering unlock password, also force the user to re-auth with Alby and check that they auth with the same Alby account as the user in the NWC DB and only then create the session cookie - otherwise show account mismatch screen with instructions

~~Also make sure Alby account is connected right at the beginning (before node is created)~~

### PR DESCRIPTION
**Pull request by im-adithya**
Fixes #231

Adds `sessionCookieOAuthKey` to the session cookie

### CODE REVIEW

File: alby/alby_http_service.go [hunk 1]
```diff
@@ -15,6 +17,11 @@ type AlbyHttpService struct {
 	appConfig    *config.AppConfig
 }
 
+const (
+	sessionCookieName     = "session"
+	sessionCookieOAuthKey = "oauthenticated"
```
Comment (rolznz) [inline alby/alby_http_service.go:22]: value seems wrong here

Comment (im-adithya) [reply alby/alby_http_service.go:22]: We don't want it to conflict with the `sessionCookieAuthKey` right? I'm storing both in the same cookie (or you didn't see the `O`?)

File: alby/alby_oauth_service.go [hunk 2]
```diff
@@ -72,24 +72,27 @@ func (svc *albyOAuthService) CallbackHandler(ctx context.Context, code string) e
...
-		if err != nil {
-			svc.logger.WithError(err).Error("Failed to fetch user me")
-			// remove token so user can retry
-			svc.config.SetUpdate(accessTokenKey, me.Identifier, "")
-			return err
-		}
-
 		svc.config.SetUpdate(userIdentifierKey, me.Identifier, "")
+	} else {
+		if existingUserIdentifier != me.Identifier {
```
Comment (rolznz) [inline alby/alby_oauth_service.go:93]: this conditional seems back to front. We should compare if the new value is != the old value, right?

File: alby/alby_oauth_service.go [hunk 3]
```diff
@@ -72,24 +72,27 @@ func (svc *albyOAuthService) CallbackHandler(ctx context.Context, code string) e
...
+			return errors.New("alby identifier does not match")
```
Comment (rolznz) [inline alby/alby_oauth_service.go:94]: does the user have enough info to know what to do here?

File: alby/alby_oauth_service.go (hunk 2 above)
Comment (rolznz) [reply alby/alby_oauth_service.go:93]: Also, the token is saved now. Even if this returns an error, the app will be logged in?

File: alby/alby_oauth_service.go (hunk 3 above)
Comment (rolznz) [reply alby/alby_oauth_service.go:94]: Sounds good. Maybe "Alby hub is connected to a different alby account. Please log out of your Alby Account at getalby.com and try again."?

File: alby/alby_oauth_service.go (hunk 2 above)
Comment (im-adithya) [reply alby/alby_oauth_service.go:93]: Resolved by removing newly set token

**Review by rolznz — APPROVED** [review]
utACK

**Pull request merged**